pkg/network/mycelium/mycelium.go

```diff
@@ -0,0 +1,235 @@
+package mycelium
+
+import (
+	"crypto/ed25519"
+	"crypto/sha512"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/oasisprotocol/curve25519-voi/primitives/x25519"
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/zos/pkg/network/namespace"
+	"github.com/threefoldtech/zos/pkg/zinit"
+)
+
+const (
+	tunName      = "my0"
+	myBin        = "mycelium"
+	zinitService = "mycelium"
+	confPath     = "/tmp/mycelium_priv_key.bin"
+	MyListenTCP  = 9651
```

Comment:
you will have to add this to the list of reserved ports of the system

Reply:
it's added to reserved ports [here](https://github.com/threefoldtech/zos/pull/2342/files/29253a863afba63ab9b040e4e464e26a970bf775#diff-4e15b6f2760eb52c8292eadf92ce4a6ba3afcf08d34b528d56081b9bc1623554R127)